tests/test_sim_result.py

```diff
@@ -36,6 +36,121 @@ def test_pickle(self):
+    
+    def test_index(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        # other_arg = {2.5:1}
+        # other_arg = {1:2.5}
+        other_arg = 2.5
+
+        self.assertEqual(result.index(other_arg), 1)
+
+    def test_pop(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+
+        result.pop(2) # Test specified index
+        self.assertEqual(result.data, [0.0, 2.5, 7.5, 10.0])
+        result.pop() # Test default index -1 (last element)
+        self.assertEqual(result.data, [0.0, 2.5, 7.5])
+        self.assertRaises(IndexError, result.pop, 5) # Test specifying an invalid index value
+        self.assertRaises(IndexError, result.pop, 3)
+        self.assertRaises(TypeError, result.pop, "5") # Test specifying an invalid index type
+        self.assertRaises(TypeError, result.pop, [0,1])
+        self.assertRaises(TypeError, result.pop, {})
+        self.assertRaises(TypeError, result.pop, set())
+        self.assertRaises(TypeError, result.pop, 1.5)
+        # self.assertRaises(TypeError, result.pop, True) # This removes index 1 because boolean True, is this okay?
+
+    def test_remove(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+
+        # print(result.times) # [0, 1, 2, 3, 4]
+        # print(result.data)  # [0.0, 2.5, 5.0, 7.5, 10.0]
+        result.remove(0) # Test specified index
+        self.assertEqual(result.times, [1, 2, 3, 4])
+        self.assertEqual(result.data, [2.5, 5.0, 7.5, 10.0])
+
+        # result.remove(1) # Test specified index
+        # self.assertEqual(result.times, [2, 3, 4]) # Passes
+        # self.assertEqual(result.data, [2.5, 5.0, 7.5, 10.0]) # wrong behavior? ValueError 
+
+        # because ValueError, any type can be passed to be removed
+        # self.assertRaises(TypeError, result.remove, ) # Test no index specified
+        # self.assertRaises(TypeError, result.remove, "5") # Tests specifying an invalid index type
+        # self.assertRaises(TypeError, result.remove, [0,1])
+        # self.assertRaises(TypeError, result.remove, {})
+        # self.assertRaises(TypeError, result.remove, set())
+        # self.assertRaises(TypeError, result.remove, 1.5)
+
+        # self.assertRaises(IndexError, result.remove, 5) # Test specifying an invalid value
+        # self.assertRaises(IndexError, result.remove, 3)
+        # self.assertRaises(TypeError, result.remove, True) # This removes index 1 because boolean True, is this okay?
+
```

Comment:
I also recommend you check the not implemented functions. Just make sure that they raise the expected exception